library/private/options_tools.h.in

```diff
@@ -23,9 +24,36 @@ std::string trim(std::string_view strv)
   return std::string(strv);
 }
 
+//----------------------------------------------------------------------------
+/** Call `std::stod` and throw `std::invalid_argument` if the whole string has not been consumed. */
```

Comment:
maybe document this in options.h as the exception output will ba a bit non-descriptive from a libf3d user POV

Reply:
`stod` already throws its own `std::invalid_argument` in some cases, we just add an additional one here and it should be caught by the templated `parse` functions and not bubble up to the users.
Also this function doesn't have to be exposed to the outside actually (but I don't know what that implies and how to deal with it)

Reply:
Indeed, ill need to doc that in the .md file though